hail/python/hailtop/utils/utils.py

```diff
@@ -33,42 +33,42 @@ def async_to_blocking(coro):
         thread_pool, lambda: fun(*args, **kwargs))
 
 
-async def gather(*pfs, parallelism=10, return_exceptions=False):
-    gatherer = AsyncThrottledGather(*pfs,
+async def bounded_gather(*aws, parallelism=10, return_exceptions=False):
+    gatherer = AsyncThrottledGather(aws,
                                     parallelism=parallelism,
                                     return_exceptions=return_exceptions)
     return await gatherer.wait()
 
 
 class AsyncThrottledGather:
-    def __init__(self, *pfs, parallelism=10, return_exceptions=False):
-        self.count = len(pfs)
+    def __init__(self, aws, parallelism=10, return_exceptions=False):
+        self.count = len(aws)
         self.n_finished = 0
 
         self._queue = asyncio.Queue()
         self._done = asyncio.Event()
         self._return_exceptions = return_exceptions
 
-        self._results = [None] * len(pfs)
+        self._results = [None] * len(aws)
         self._errors = []
 
         self._workers = []
         for _ in range(parallelism):
             self._workers.append(asyncio.ensure_future(self._worker()))
 
-        for i, pf in enumerate(pfs):
-            self._queue.put_nowait((i, pf))
+        for i, aw in enumerate(aws):
+            self._queue.put_nowait((i, aw))
 
     def _cancel_workers(self):
         for worker in self._workers:
             worker.cancel()
 
     async def _worker(self):
         while True:
-            i, pf = await self._queue.get()
+            i, aw = await self._queue.get()
```

Comment:
Did you test whether this code generates the error I was seeing where all of the awaitables hadn't been awaited?

Reply:
You're right.  I think the code might be best as is.  Sorry for doubting!  It looks like there is no way to signal a coroutine isn't going to be waited upon. The usual thing is to call create_task/ensure_future, but that schedules it for execution.  Either way, the partial function is probably more efficient.  I guess the lesson for me is: don't use awaitables for deferred execution.